dlt/destinations/impl/synapse/synapse.py

```diff
@@ -0,0 +1,175 @@
+            self, table_name, new_columns, generate_alter
+        )
+        if not generate_alter:
+            table_index_type_attr = TABLE_INDEX_TYPE_TO_SYNAPSE_ATTR[table_index_type]
+            sql_result = [_sql_result[0] + f"\n WITH ( {table_index_type_attr} );"]
+        else:
+            sql_result = _sql_result
+        return sql_result
+
+    def _get_columstore_valid_columns(
+        self, columns: Sequence[TColumnSchema]
+    ) -> Sequence[TColumnSchema]:
+        return [self._get_columstore_valid_column(c) for c in columns]
+
+    def _get_columstore_valid_column(self, c: TColumnSchema) -> TColumnSchema:
+        """
+        Returns TColumnSchema that maps to a Synapse data type that can participate in a columnstore index.
+
+        varchar(max), nvarchar(max), and varbinary(max) are replaced with
+        varchar(n), nvarchar(n), and varbinary(n), respectively, where
+        n equals the user-specified precision, or the maximum allowed
+        value if the user did not specify a precision.
+        """
+        varchar_source_types = [
+            sct
+            for sct, dbt in MsSqlTypeMapper.sct_to_unbound_dbt.items()
+            if dbt in ("varchar(max)", "nvarchar(max)")
+        ]
+        varbinary_source_types = [
+            sct
+            for sct, dbt in MsSqlTypeMapper.sct_to_unbound_dbt.items()
+            if dbt == "varbinary(max)"
+        ]
+        if c["data_type"] in varchar_source_types and "precision" not in c:
+            return {**c, **{"precision": VARCHAR_MAX_N}}
+        elif c["data_type"] in varbinary_source_types and "precision" not in c:
+            return {**c, **{"precision": VARBINARY_MAX_N}}
+        return c
+
+    def _create_replace_followup_jobs(
+        self, table_chain: Sequence[TTableSchema]
+    ) -> List[NewLoadJob]:
+        if self.config.replace_strategy == "staging-optimized":
+            return [SynapseStagingCopyJob.from_table_chain(table_chain, self.sql_client)]
+        return super()._create_replace_followup_jobs(table_chain)
+
+    def get_load_table(self, table_name: str, staging: bool = False) -> TTableSchema:
+        table = super().get_load_table(table_name, staging)
+        if table is None:
+            return None
+        if table_name in self.schema.dlt_table_names():
+            # dlt tables should always be heap tables, regardless of the user
+            # configuration. Why? "For small lookup tables, less than 60 million rows,
+            # consider using HEAP or clustered index for faster query performance."
+            # https://learn.microsoft.com/en-us/azure/synapse-analytics/sql-data-warehouse/sql-data-warehouse-tables-index#heap-tables
+            table["table_index_type"] = "heap"
+        if table["table_index_type"] is None:
+            table["table_index_type"] = self.config.default_table_index_type
+        return table
+
+    def get_storage_table_index_type(self, table_name: str) -> TTableIndexType:
```

Comment:
I think you can move this to the tests if it is not used in production code

Reply:
Done.